lib/zendesk_apps_support/validations/requests.rb

```diff
@@ -1,57 +1,108 @@
 # frozen_string_literal: true
 
-require 'uri'
+# TODO: localize string to indicate unrecognised urls detected
+# package.warnings << validation_message(
+#   'txt.apps.admin.warning.app_build.unrecognised_urls_detected',
+#   urls: unrecognised_urls.join(', '),
+#   file: file.relative_path
+# )
+
 require 'ipaddress_2'
 
 module ZendeskAppsSupport
   module Validations
     module Requests
       class << self
-        HTTP_REQUEST_METHODS = %w[GET HEAD POST PUT DELETE CONNECT OPTIONS TRACE].freeze
+        HTTP_URI = /http[^s].*/i
 
-        REQUEST_CALLS = [
-          /\w+\.request\(['"](.*)['"]/i, # ZAF request
-          /(?:\$|jQuery)\.(?:ajax|get|post|getJSON)\(['"](.*)['"]/i, # jQuery request
-          /\w+\.open\(['"](?:#{Regexp.union(HTTP_REQUEST_METHODS).source})['"],\s?['"](.*)['"]/i, # XMLHttpRequest
-          /fetch\(['"](.*)['"]/i # fetch
+        COMMON_URLS = %r{\"?\'?(http(s)?:\/\/|w{3}\.)\w}i
+        ALLOWLISTED_URLS = %w[
+          zendesk.com/api/v2/ support.zendesk.com https://assets.zendesk.com/apps/sdk/2.0/zaf_sdk.js
+          garden.zendesk.com https://github.com/zendesk/zendesk_app_framework_sdk
+          assets.zendesk.com www.apache.org https://ajax.googleapis.com https://cdn.jsdelivr.net
         ].freeze
+        ALLOWLISTED_URLS_REGEXP = Regexp.new(ALLOWLISTED_URLS.join('|'), Regexp::IGNORECASE)
+
+        IP_ADDRESS_BLOCKS = /\d{,2}|1\d{2}|2[0-4]\d|25[0-5]/
+        IP_ADDRESS = /\A#{IP_ADDRESS_BLOCKS}\.#{IP_ADDRESS_BLOCKS}\.#{IP_ADDRESS_BLOCKS}\.#{IP_ADDRESS_BLOCKS}\z/
 
         def call(package)
           errors = []
           files = package.js_files + package.html_files
 
           files.each do |file|
-            contents = file.read
-            REQUEST_CALLS.each do |request_pattern|
-              request = contents.match(request_pattern)
-              next unless request
-              uri = URI(request.captures[0])
-              if uri.scheme == 'http'
-                package.warnings << I18n.t('txt.apps.admin.warning.app_build.insecure_http_request',
-                                           uri: uri,
-                                           file: file.relative_path)
+            file_content = file.read.split(' ').map { |word| word.gsub(/[\"\']/, '') }
+
+            http_protocol_urls = search_in_file(HTTP_URI, file_content)
+            if http_protocol_urls.any?
+              package.warnings << validation_message(
+                'txt.apps.admin.warning.app_build.insecure_http_request',
+                uri: http_protocol_urls.join(', '),
+                file: file.relative_path
+              )
+            end
+
+            url_addresses = search_in_file(COMMON_URLS, file_content)
+            if url_addresses.any?
+              unrecognised_urls = find_unrecognised_urls(url_addresses)
+              if unrecognised_urls.any?
+                package.warnings << unrecognised_urls_warning(file.relative_path, unrecognised_urls)
```

Comment:
How come we're adding this new validation? Even if we do want to start warning on this, I wonder if it should be split out into a separate PR/story so the bugfixes and refactoring here can be reviewed independently

Reply:
That is an excellent idea! I will put this in a separate PR.

As discussed in person, the reason why I put it in here was because I saw this in Kari's work in AppsApproval. 

Running this by Chris, he mentioned that this can be valuable in the apps upload process in ZAM for Eric. Therefore, I will think about how I can move this there in a seperate ticket.